db/migrate/20190722082648_add_derived_estimated_hours_to_work_packages.rb

```diff
@@ -0,0 +1,112 @@
+class AddDerivedEstimatedHoursToWorkPackages < ActiveRecord::Migration[5.2]
+  class WorkPackageWithRelations < ActiveRecord::Base
+    self.table_name = WorkPackage.table_name
+
+    scope :with_children, ->(*args) do
+      rel = Relation.table_name
+      wp = WorkPackage.table_name
+
+      query = "EXISTS (SELECT 1 FROM #{rel} WHERE #{rel}.from_id = #{wp}.id AND #{rel}.hierarchy > 0 LIMIT 1)"
+
+      where(query)
+    end
+  end
+
+  def change
+    add_column :work_packages, :derived_estimated_hours, :float
+    add_column :work_package_journals, :derived_estimated_hours, :float
+
+    reversible do |change|
+      change.up do
+        WorkPackage.transaction do
+          migrate_to_derived_estimated_hours!
+        end
+      end
+    end
+  end
+
+  # Before this migration all work packages who have children had their
+  # estimated hours set based on their children through the UpdateAncestorsService.
+  #
+  # We move this value to the derived_estimated_hours column and clear
+  # the estimated_hours column. In the future users can estimte the time
+  # for parent work pacages separately there while the UpdateAncestorsService
+  # only touches the derived_estimated_hours column.
+  def migrate_to_derived_estimated_hours!
+    last_id = Journal.order(id: :desc).limit(1).pluck(:id).first
+    wp_journals = WorkPackageJournal.table_name
+    author = User.system
+    notes = "_Estimated Hours migrated to Derived Estimated Hours_"
```

Comment:
Spelling? Do we actually want to have the names starting with uppercase in every word? How about 
`_'Estimated hours' changed to 'Derived estimated hours'_`